doc/release-notes-pr12892.md

```diff
@@ -0,0 +1,28 @@
+'label' API for wallet
+----------------------
+
+A new 'label' API has been introduced for the wallet. This is intended as a
```

Comment:
This is great! There are a lot of changes I would suggest but I thought it would be easier to just make them in a wiki page:

https://github.com/bitcoin-core/bitcoin-devwiki/wiki/12892-Label-API-release-notes

The main change is trying to organize the RPC information in a table.

Feel free to delete the wiki page and move anything here.